platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleStateRoot.java

```diff
@@ -356,13 +357,27 @@ public ReadableStates getReadableStates(@NonNull String serviceName) {
     @Override
     @NonNull
     public WritableStates getWritableStates(@NonNull final String serviceName) {
+        logger.info(STARTUP.getMarker(), "Getting writable states for service {}", serviceName);
+        logLimitedStackTrace();
         throwIfImmutable();
         return writableStatesMap.computeIfAbsent(serviceName, s -> {
             final var stateMetadata = services.getOrDefault(s, Map.of());
             return new MerkleWritableStates(serviceName, stateMetadata);
         });
     }
 
+    private void logLimitedStackTrace() {
```

Comment:
this could introduce a performance issue, I don't know how often this is called. I think the best way to go is to not allow access to the mutable state by anyone other than the consensus thread

Reply:
This is only for information gathering purposes. This code will not get into prod.

Reply:
there will be a PR to remove it?

Reply:
No, this PR is just for experiments :)